src/Symmetric/Openssl.php

```diff
@@ -0,0 +1,552 @@
+ * PLEASE USE Zend\Crypt\BlockCipher instead!
+ */
+class Openssl implements SymmetricInterface
+{
+    const DEFAULT_PADDING = 'pkcs7';
+
+    /**
+     * Key
+     *
+     * @var string
+     */
+    protected $key;
+
+    /**
+     * IV
+     *
+     * @var string
+     */
+    protected $iv;
+
+    /**
+     * Encryption algorithm
+     *
+     * @var string
+     */
+    protected $algo = 'aes';
+
+    /**
+     * Encryption mode
+     *
+     * @var string
+     */
+    protected $mode = 'cbc';
+
+    /**
+     * Padding
+     *
+     * @var Padding\PaddingInterface
+     */
+    protected $padding;
+
+    /**
+     * Padding plugins
+     *
+     * @var Interop\Container\ContainerInterface
+     */
+    protected static $paddingPlugins = null;
+
+    /**
+     * Supported cipher algorithms
+     *
+     * @var array
+     */
+    protected $supportedAlgos = [
+        'aes'      => 'AES-256',
+        'blowfish' => 'BF',
+        'des'      => 'DES',
+        'camellia' => 'CAMELLIA-256',
+        'cast5'    => 'CAST5',
+        'seed'     => 'SEED'
+    ];
+
+    /**
+     * Supported encryption modes
+     *
+     * @var array
+     */
+    protected $supportedModes = [
+        'cbc',
+        'cfb',
+        'ofb',
+        'ecb'
+    ];
+
+    /**
+     * Block sizes (in bytes) for each supported algorithm
+     *
+     * @var array
+     */
+    protected $blockSizes = [
+        'aes'      => 16,
+        'blowfish' => 8,
+        'des'      => 8,
+        'camellia' => 16,
+        'cast5'    => 8,
+        'seed'     => 16
+    ];
+
+    /**
+     * Key sizes (in bytes) for each supported algorithm
+     *
+     * @var array
+     */
+    protected $keySizes = [
+        'aes'      => 32,
+        'blowfish' => 56,
+        'des'      => 8,
+        'camellia' => 32,
+        'cast5'    => 16,
+        'seed'     => 16
+    ];
+
+    /**
+     * Constructor
+     *
+     * @param  array|Traversable                  $options
+     * @throws Exception\RuntimeException
+     * @throws Exception\InvalidArgumentException
+     */
+    public function __construct($options = [])
+    {
+        if (!extension_loaded('openssl')) {
+            throw new Exception\RuntimeException(
+                'You cannot use ' . __CLASS__ . ' without the OpenSSL extension'
+            );
+        }
+        $this->setOptions($options);
+        $this->setDefaultOptions($options);
+    }
+
+    /**
+     * Set default options
+     *
+     * @param  array $options
+     * @return void
+     */
+    public function setOptions($options)
+    {
+        if (!empty($options)) {
+            if ($options instanceof Traversable) {
+                $options = ArrayUtils::iteratorToArray($options);
+            } elseif (!is_array($options)) {
+                throw new Exception\InvalidArgumentException(
+                    'The options parameter must be an array, a Zend\Config\Config object or a Traversable'
```

Comment:
Drop the reference to `Zend\Config\Config`, as it falls under the `Traversable` umbrella.